### PR TITLE
Add access log

### PR DESCRIPTION
**Pull request by xiaoliu10**
1. add access log filter

### CODE REVIEW

File: pkg/filter/accesslog/access_log_filter.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+	// console
+	Console = "console"
+)
+
+func init() {
+	extension.SetFilterFunc(constant.AccessLogFilter, Filter())
+}
+
+func Filter() context.FilterFunc {
```
Comment (zhangshen023) [inline pkg/filter/accesslog/access_log_filter.go:53]: Func Filter may be un exported

File: pkg/filter/accesslog/access_log_filter.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	return func(c context.Context) {
+		alc := config.GetBootstrap().StaticResources.AccessLogConfig
+		if alc.Enable {
```
Comment (zhangshen023) [inline pkg/filter/accesslog/access_log_filter.go:56]: `if !alc. Enable{
    return
}`

File: pkg/filter/accesslog/access_log_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accesslog
+
+import (
+	"github.com/stretchr/testify/assert"
```
Comment (cityiron) [inline pkg/filter/accesslog/access_log_test.go:21]: split

File: pkg/model/log.go [hunk 4]
```diff
@@ -24,6 +24,11 @@ type AccessLog struct {
 	Config interface{}     `yaml:"config" json:"config" mapstructure:"config"`
 }
 
+type AccessLogConfig struct {
+	Enable     bool   `yaml:"enable" json:"enable" mapstructure:"enable"`
+	OutPutPath string `yaml:"outPutPath" json:"outPutPath" mapstructure:"outPutPath"`
+}
+
 // AccessLogFilter
```
Comment (cityiron) [inline pkg/model/log.go:32]: add more info tell what is AccessLogFilter

File: pkg/filter/accesslog/access_log_filter.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+			start := time.Now()
+			c.Next()
+			latency := time.Now().Sub(start)
+			// build access_log message
+			accessLogMsg := buildAccessLogMsg(c, latency)
+			if len(alc.OutPutPath) == 0 || alc.OutPutPath == Console {
```
Comment (cityiron) [inline pkg/filter/accesslog/access_log_filter.go:62]: like zhangshen023's suggest,  
```go
if xx {
  return
}

xxx
```

**Review by cityiron — APPROVED** [review]

File: pkg/filter/accesslog/access_log_filter.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+func Filter() context.FilterFunc {
+	return func(c context.Context) {
+		alc := config.GetBootstrap().StaticResources.AccessLogConfig
+		if alc.Enable {
+			start := time.Now()
+			c.Next()
+			latency := time.Now().Sub(start)
+			// build access_log message
+			accessLogMsg := buildAccessLogMsg(c, latency)
```
Comment (zhangshen023) [inline pkg/filter/accesslog/access_log_filter.go:61]: Creating `accessLogMsg`  once

**Review by zhangshen023 — COMMENTED** [review]
@xiaoliu10

File: pkg/filter/accesslog/access_log_filter.go (hunk 6 above)
Comment (xiaoliu10) [reply pkg/filter/accesslog/access_log_filter.go:61]: i don't quite understand what that means

**Review by williamfeng323 — APPROVED** [review]

**Review by yqxu — APPROVED** [review]

**Pull request merged**